### PR TITLE
Cisco duo

### PR DESCRIPTION
**Pull request by surajkadam**
Alert Logic AWS Based API Poll (PAWS) Log Collector Library.

### Solution Description
This repository contains the AWS JavaScript Lambda function and CloudFormation Template (CFT) for deploying a log collector in AWS which will poll a 3rd party service API to collect and forward logs to the Alert Logic CloudInsight backend services.

### Acceptance Criteria for Contributors
- [x] No errors / warnings on build (including linter)
- [x] 94% automated test coverage
- [x] Source layout followed from other collectors
- [x] No unnecessary code (debug logs, redundant comments, unused blocks of code, etc.)
- [x] Logging of main steps in the collector’s code
- [x] Logging of any requests that cause the collector to fail / throw an exception, with reason for failure / debugging info
- [ ] API throttling errors are understood and and properly handled
- [x] Registration/update/deregistration validated
- [x] CloudFormation template specific to setting up the collector, with documentation of any template parameters
- [x] Collector README, including any set up caveats (both when installed in AL account and in customer’s account) – the user/team should be able to successfully set up, update, remove a collector following the README only
- [x] Demo of set up, update, removal of a collector, with data shown to be correctly parsed in AL console search
- [x] Links to API documentation

### CODE REVIEW

File: collectors/ciscoduo/collector.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                object: curState.object,
+                mintime: curState.mintime,
+                maxtime: curState.maxtime,
+                nextPage: nextPage,
+                poll_interval_sec: 1
+            };
+        } else {
+            return {
+                object: curState.object,
+                mintime: nextPage,
```
[the remark this answers is not in the document]
Comment (surajkadam) [reply collectors/ciscoduo/collector.js:157]: we don't have the next page concept for this API. Actually we passed last log Unix timestamps + 1. we will Pull the next logs bypassing these timestamps to mintime.

File: collectors/ciscoduo/cfn/ciscoduo-collector.template [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        },
+        "CiscoduoSecret": {
+            "Description": "Cisco Duo Secret key for authentication.",
+            "Type": "String",
+            "NoEcho": true
+        },
+        "CiscoduoObjectNames": {
+            "Description": "Define Object names. Please pass JSON formatted list.",
+            "Type": "String",
+            "Default": "[\"Authentication\", \"Administrator\",\"Telephony\", \"OfflineEnrollment\"]"
```
[the remark this answers is not in the document]
Comment (surajkadam) [reply collectors/ciscoduo/cfn/ciscoduo-collector.template]: Done

File: collectors/ciscoduo/collector.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (surajkadam) [reply collectors/ciscoduo/collector.js:157]: As per document "Returns a list of administrator log events. Only the 1000 earliest events will be returned; you may need to call this multiple times with mintime to page through the entire log. Requires "Grant read log" API permission."

[the remark this answers is not in the document]
Comment (JDragovichAlertLogic) [reply collectors/ciscoduo/collector.js:157]: Fair enough. Might want to put a comment there about that and point to the docs.

**Pull request merged**